src/lib/components/consignments/components/boolean_button.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from "react"
```

Comment:
I was thinking that the `onPress` would end up changing a parent's state that would eventually trickle down into the `selected` state, so handling the transition in the component didn't feel right. If it was doing animation, I'd definitely agree though. Then it might be a onChange kind of affair.

WRT name: agreed